lib/src/config_provider/config.dart

```diff
@@ -215,363 +219,699 @@ class Config {
+            ],
+            transform: (node) => compilerOptsExtractor(node.value),
+          ),
+          defaultValue: (node) => <String>[],
+          resultOrDefault: (node) => _compilerOpts = node.value as List<String>,
+        ),
+        FixedMapEntry(
+            key: strings.compilerOptsAuto,
+            valueConfigSpec: FixedMapConfigSpec(
+              entries: [
+                FixedMapEntry(
+                  key: strings.macos,
+                  valueConfigSpec: FixedMapConfigSpec(
+                    entries: [
+                      FixedMapEntry(
+                        key: strings.includeCStdLib,
+                        valueConfigSpec: BoolConfigSpec(),
+                        defaultValue: (node) => true,
+                      )
+                    ],
+                  ),
+                )
+              ],
+              transform: (node) => CompilerOptsAuto(
+                macIncludeStdLib: ((node.value)[strings.macos]
+                    as Map?)?[strings.includeCStdLib] as bool,
+              ),
+              result: (node) => _compilerOpts.addAll(
+                  (node.value as CompilerOptsAuto).extractCompilerOpts()),
+            )),
+        FixedMapEntry(
+          key: strings.libraryImports,
+          valueConfigSpec:
+              DynamicMapConfigSpec<String, Map<String, LibraryImport>>(
+            keyValueConfigSpecs: [
+              (keyRegexp: ".*", valueConfigSpec: StringConfigSpec()),
+            ],
+            customValidation: _libraryImportsPredefinedValidation,
+            transform: (node) => libraryImportsExtractor(node.value.cast()),
+          ),
+          defaultValue: (node) => <String, LibraryImport>{},
+          resultOrDefault: (node) =>
+              _libraryImports = (node.value) as Map<String, LibraryImport>,
+        ),
+        FixedMapEntry(
+            key: strings.functions,
+            valueConfigSpec: FixedMapConfigSpec(
+              entries: [
+                ..._includeExcludeProperties(),
+                ..._renameProperties(),
+                ..._memberRenameProperties(),
+                FixedMapEntry(
+                  key: strings.symbolAddress,
+                  valueConfigSpec: _includeExcludeObject(),
+                  defaultValue: (node) => Includer.excludeByDefault(),
+                ),
+                FixedMapEntry(
+                  key: strings.exposeFunctionTypedefs,
+                  valueConfigSpec: _includeExcludeObject(),
+                  defaultValue: (node) => Includer.excludeByDefault(),
+                ),
+                FixedMapEntry(
+                  key: strings.leafFunctions,
+                  valueConfigSpec: _includeExcludeObject(),
+                  defaultValue: (node) => Includer.excludeByDefault(),
+                ),
+                FixedMapEntry(
+                  key: strings.varArgFunctions,
+                  valueConfigSpec: _functionVarArgsSchema(),
```

Comment:
Schema -> ConfigSpec ? Please do a ctrl+f on Schema to find all the old naming.

Since it's a private method it doesn't matter that much, but still cleaner to clean it up.